lsr-role2collection.py

```diff
@@ -0,0 +1,418 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# (c) 2020 Matt Martz <matt@sivel.net>
+# GNU General Public License v3.0+
+#     (see https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Usage:
+# lsr-role2collection.py [--namespace NAMESPACE]
+#                        [--collection COLLECTION]
+#                        --src-path SRC_PATH
+#                        --dest-path DEST_PATH
+#                        --role ROLE
+#                        [-h]
+
+import argparse
+import os
+import re
+import fnmatch
+from shutil import copytree, copy2, ignore_patterns
+
+from pathlib import Path
+
+ROLE_DIRS = (
+    'defaults',
+    'files',
+    'handlers',
+    'meta',
+    'tasks',
+    'templates',
+    'vars',
+)
+
+PLUGINS = (
+    'action_plugins',
+    'become_plugins',
+    'cache_plugins',
+    'callback_plugins',
+    'cliconf_plugins',
+    'connection_plugins',
+    'doc_fragments',
+    'filter_plugins',
+    'httpapi_plugins',
+    'inventory_plugins',
+    'library',
+    'lookup_plugins',
+    'module_utils',
+    'netconf_plugins',
+    'shell_plugins',
+    'strategy_plugins',
+    'terminal_plugins',
+    'test_plugins',
+    'vars_plugins'
+)
+
+TESTS = (
+    'tests',
+    'molecule',
+    '.travis',
+)
+
+DOCS = (
+    'docs',
+    'design_docs',
+)
+
+ALL_DIRS = ROLE_DIRS + PLUGINS + TESTS + DOCS
+
+IMPORT_RE = re.compile(
+    br'(\bimport) (ansible\.module_utils\.)(\S+)(.*)$',
+    flags=re.M
+)
+FROM_RE = re.compile(
+    br'(\bfrom) (ansible\.module_utils\.?)(\S+)? import (\(*(?:\n|\r\n)?)(.+)$',
+    flags=re.M
+)
```

Comment:
@pcahyna, regarding the change made in https://github.com/linux-system-roles/storage/pull/142,
```
- from ansible.module_utils.size import Size
+ from ansible.module_utils.storage.size import Size
```
Both from ~ import ~ lines match the FROM_RE regular expression.

Pre pr/142: the from ~ import ~ is converted to:
```
plugins/modules/bsize.py:from ansible_collections.redhat.system_roles.plugins.module_utils.size import Size
plugins/modules/find_unused_disk.py:from ansible_collections.redhat.system_roles.plugins.module_utils.size import Size
```
and size.py is located at:
`plugins/module_utils/size.py`

Post pr/142: the from ~ import ~ is converted to:
```
plugins/modules/bsize.py:from ansible_collections.redhat.system_roles.plugins.module_utils.storage.size import Size
plugins/modules/find_unused_disk.py:from ansible_collections.redhat.system_roles.plugins.module_utils.storage.size import Size
```
and size.py is located at:
`plugins/module_utils/storage/size.py`

Reply:
@t-woerner , is there a way to keep the same module_utils import path with and without collections and not have to rewrite the Python sources?

Reply:
For a collection all plugins (module, module_utils, action_plugins,...) need to be moved or cpoied into the collection global plugins directory. Plugins within role directories are not loaded.

Reply:
All plugins need to have a proper prefix and a unique name as they have to be located in the global plugins directory.